src/sidenav/sidenav-item.tsx

```diff
@@ -0,0 +1,68 @@
+import { useContext, useEffect, useState } from "react";
+import {
+    Container,
+    DefaultButton,
+    IconContainer,
+    TitleText,
+} from "./sidenav-item.styles";
+import { SidenavItemProps } from "./types";
+import { SidenavContext } from "./sidenav-context";
+
+export const SidenavItem = ({
+    icon,
+    title,
+    onClick,
+    ...otherProps
+}: SidenavItemProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const id = otherProps.id || title.toLowerCase().replaceAll(" ", "-");
+    const { selectedItem, setSelectedItem } = useContext(SidenavContext);
+    const [selected, setSelected] = useState<boolean>();
```

Comment:
Actually if selectedItem is already an id, i don't see a need to have another state value for this. Can just use selectedItem?